packages/providers/src/fuel-graphql-request.ts

```diff
@@ -0,0 +1,154 @@
+/* 
+
+  Replaces `graphql-request` and `graphql-tag` dependencies with a simple
+  `fetch` implementation, reducing the final bundle size by 15.68 kB.
+
+  More:
+    https://github.com/FuelLabs/fuels-ts/discussions/1592#discussioncomment-8003515
+
+*/
+
+import { FuelError } from '@fuel-ts/errors';
+import type { DocumentNode } from 'graphql';
+import { print } from 'graphql';
+
+function isSubscription({ definitions: defs }: DocumentNode) {
+  const opDef = defs.find((op) => op.kind === 'OperationDefinition');
+
+  if (opDef && 'operation' in opDef) {
+    return opDef?.operation === 'subscription';
+  }
+
+  return false;
+}
+
+class FuelGraphqlSubscriber implements AsyncIterator<unknown> {
+  private stream!: ReadableStreamDefaultReader<Uint8Array>;
+  private static textDecoder = new TextDecoder();
+
+  private static parseBytesStream(
+    bytes: Uint8Array | undefined
+  ): Record<string, unknown> | FuelError | undefined {
+    if (bytes === undefined) {
+      return undefined;
+    }
+
+    const text = this.textDecoder.decode(bytes);
+
+    // https://github.com/FuelLabs/fuel-core/blob/e1e631902f762081d2124d9c457ddfe13ac366dc/crates/fuel-core/src/graphql_api/service.rs#L247
+    if (text === 'keep-alive-text') {
+      return undefined;
+    }
+
+    const { data, errors } = JSON.parse(text.split('data:')[1]);
+
+    if (Array.isArray(errors)) {
+      return new FuelError(
+        FuelError.CODES.INVALID_REQUEST,
+        errors.map((err) => err.message).join('\n\n')
+      );
+    }
+
+    return data as Record<string, unknown>;
+  }
+
+  public constructor(
+    private options: {
+      url: string;
+      request: RequestInit;
+      fetchFn: typeof fetch;
+    }
+  ) {}
+
+  private async setStream() {
+    const { url, request, fetchFn } = this.options;
+    const response = await fetchFn(`${url}-sub`, {
```

Comment:
Does this URL modification assume that the URL was given in a very specific and unique format? 

Can there be cases where this URL can be given in a way that will make this fetch result in an error?

Reply:
The `-sub` prefix is added by the fuel node [see here](https://github.com/FuelLabs/fuel-core/blob/e1e631902f762081d2124d9c457ddfe13ac366dc/crates/fuel-core/src/graphql_api/service.rs#L191) so it can't be modified. The fuel node serves subscriptions over `/graphql-sub`.